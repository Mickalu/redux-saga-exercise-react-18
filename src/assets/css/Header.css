.hr-title {
  opacity: 1;

}

.hr-title::after {
  opacity: 0;
}

.row{
  margin-left: 0%;
  margin-right: 0%;
}
