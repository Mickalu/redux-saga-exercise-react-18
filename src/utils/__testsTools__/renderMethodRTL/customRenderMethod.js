import React from "react";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";

import { store } from "../../../store";

const getByTextCustom = (text) => (
  screen.getByText(text)
);

const queryAllByRoleCustom = (role, options) => (
  screen.queryAllByRole(role, options)
);

const queryByTestIdCustom = (testId) => (
  screen.queryByTestId(testId)
);

const getByTestIdCustom = (testId) => (
  screen.getByTestId(testId)
);

const getByDisplayValueCustom = (displayValue) => (
  screen.getByDisplayValue(displayValue)
);

const renderWithProviders = ({ children }) => (
  <Provider store={store}>
    { children }
  </Provider>
);

export const renderCustom = (ui, options = {}) => render(ui, { wrapper: renderWithProviders, ...options });

// re-export everything
export * from '@testing-library/react';

export { renderCustom as render };
export { getByTextCustom as getByText };
export { queryAllByRoleCustom as queryAllByRole };
export { queryByTestIdCustom as queryBytestId };
export { getByTestIdCustom as getByTestId };
export { getByDisplayValueCustom as getByDisplayValue };
