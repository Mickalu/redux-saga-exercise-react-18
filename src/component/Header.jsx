import React from 'react';

import "../assets/css/main.css";
import "../assets/css/Header.css";

import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Image from 'react-bootstrap/Image';

const Header = () => (
  <header>
    <Container>
      <Row>
        <div className="col-lg-12">
          <Image className="img-responsive logo-academy" alt="beer" role="presentation" src={"/img/logo.png"} />
          <div className="intro-text">
            <span className="name">Beer Game</span>
            <hr className="star-light hr-title" />
          </div>
        </div>
      </Row>
    </Container>
  </header>
);

export default Header;
