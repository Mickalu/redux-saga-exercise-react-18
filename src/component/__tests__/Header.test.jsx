import React from "react";

import { render } from "../../utils/__testsTools__/renderMethodRTL/customRenderMethod";
import Header from "../Header";

it("Header should match snapshot", () => {
  const { container }  = render(<Header />);

  expect(container).toMatchSnapshot();
});
