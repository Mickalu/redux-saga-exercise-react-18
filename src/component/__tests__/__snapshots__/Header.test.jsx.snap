// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<div>
  <header>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-12"
        >
          <img
            alt="beer"
            class="img-responsive logo-academy"
            role="presentation"
            src="/img/logo.png"
          />
          <div
            class="intro-text"
          >
            <span
              class="name"
            >
              Beer Game
            </span>
            <hr
              class="star-light hr-title"
            />
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
